icu4c/source/common/unicode/uset.h

```diff
@@ -1756,26 +1759,28 @@ class USetElementIterator {
         return uset == other.uset && c == other.c && index == other.index;
     }
 
-    /** @draft ICU 76 */
+    /** @draft ICU 77 */
     bool operator!=(const USetElementIterator &other) const { return !operator==(other); }
 
-    /** @draft ICU 76 */
-    UnicodeString operator*() const {
+    /** @draft ICU 77 */
+    std::u16string operator*() const {
         if (c >= 0) {
-            return UnicodeString(c);
+            return c <= 0xffff ?
```

Comment:
Why not use the `U_IS_BMP()` macro here too, like in the test code?

Reply:
i don't think it really matters one way or another; slight preference not to add an `include unicode/utf.h` to uset.h

Reply:
It's just odd to me to do the same thing in two different ways in the same PR for no obvious reason.

Reply:
Well, so I gave is a mild reason.
Either way the code only makes sense to someone who understands code points and UTF-16, and if they do, then both versions should be equally fine.